tensorflow/python/lib/io/file_io_test.py

```diff
@@ -195,6 +196,20 @@ def testGetMatchingFilesWhenParentDirContainsParantheses(self):
     self.assertItemsEqual(
         file_io.get_matching_files(glob_pattern), expected_match)
 
+  @unittest.expectedFailure
+  def testGetMatchingFilesWhenContainsGlob(self):
+    dir_path = os.path.join(self._base_dir, "[dir_special]")
+    file_io.create_dir(dir_path)
+    files = ["file1.txt", "file*2?].txt"]
+    for name in files:
+      file_path = os.path.join(dir_path, name)
+      file_io.FileIO(file_path, mode="w").write("testing")
+    expected_match = [os.path.join(dir_path, name) for name in files]
+    glob_pattern = os.path.join(dir_path, "*")
+    self.assertItemsEqual(
+        file_io.get_matching_files(glob_pattern), expected_match,
+        "Check #35489 https://git.io/JR8jL")
```

Comment:
Let's change this to be the direct link to the issue (e.g. `https://github.com/tensorflow/tensorflow/issue/35489`)

Reply:
Done